contributing/tests.md

```diff
@@ -7,268 +7,56 @@ describes them and gives pointers on how to run/debug them.
-
-Tip: Depending on what you are trying to do, it is likely that you need to
-rebuild the modules after the coverage tests to get to a pristine state.
+Unit tests priorities are as follows:
+
+1. Unit tests should focus on only the _unit_ of code under test.
+   - 1 method/function
+1. All dependencies should be stubbed/mocked.
+1. Unit tests should be easy to write and execute
+   - If your code is hard to test consider how it could be refactored to make it easier to test.
+1. Use only jest and do not import mochi/sinon/chai.
+1. Code coverage is one measure of how well we are unit testing our code. We should strive to cover all code paths for any touched code in the repository. Code coverage can be generated by including the [`--coverage`](https://jestjs.io/docs/cli#--coverageboolean) flag to jest when executing.
+   - Example of running jest includeing coverage generation `jest --coverage`
+
+#### How to Write Jest Unit Tests
+
+- Test files use the {fileUnderTestName}.test.ts format and should go under the same directory structure as the source in the test/jest folder.
+  - Note this is to separate from the existing mocha unit tests and enable us to track our progress for moving off mocha to jest.
+  - Example:
+    - File under test: `src/commands/auth/authParamsGatherer.ts`
+    - Test File location and name: `test/jest/commands/auth/authParamsGatherer.test.ts`
+- VSCode extensions are our friend
+  - https://marketplace.visualstudio.com/items?itemName=firsttris.vscode-jest-runner
+- Tests can be executed from the IDE, command line, or via npm script.
+  - IDE: assuming you are using the above extension, use the Run & Debug CodeLens found in test files.
+  - Command line: `npx jest` will execute all unit tests in a package by default. If the integration tests are configured for jest in the package you may execute those using `npx jest -c jest.integration.config`. There are many options for configuring the test run see the [docs](https://jestjs.io/docs/cli) or `npx jest -h` for further guidance.
+  - npm scripts:
+    - `npm run test:unit`
+    - `npm run test:integration`
+    - `npm run test`
+
+#### Jest Information
+
+- ts-jest enables us to compile on the fly so no need for compilation while writing unit tests.
+- Jest [Docs](https://jestjs.io/docs/getting-started) are very good.
+- There is a shared top level jest config file for [unit](https://github.com/forcedotcom/salesforcedx-vscode/blob/develop/config/jest.base.config.js) and [integration](https://github.com/forcedotcom/salesforcedx-vscode/blob/develop/config/jest.integration.config.js) tests. The unit test configuration is shared between the exisiting tests in the unit directory and the tests in the jest directory.
+- There is a top level jest setup script [setup-jest.js](https://github.com/forcedotcom/salesforcedx-vscode/blob/develop/scripts/setup-jest.ts) where we are stubbing the vscode module.
```

Comment:
This is pretty important, and might be worth calling out as a separate section with maybe an example?